### DISCUSSION TITLE
How to connect to a remote K8 cluster

### DISCUSSION THREAD

**Discussion by josha1971** [kubernetes-client/csharp#1054] [Q&A]
Hello! I have never done anything with k8 before so please bear with me!

I am trying to create a simple C# console app to test this nuget package out. I have access to the k8 cluster I want to use, but have only accessed it via PowerShell by logging with a user's password when the ssh in PowerShell prompts me i.e.:

> ssh root@xxx.xx.xxx.xx
> root@xxx.xx.xxx.xx's password:

From there, I can setup the proxy to localhost:8080 and start accessing the data from the various api URLs via curl requests i.e.
curl localhost:8080/api/v1/pods

I'm wondering how I can use this c# library and do the same type of thing in my console app? I see the 
var config = KubernetesClientConfiguration.BuildDefaultConfig( );
in the startup guide, but this is failing with a "YamlDotNet.Core.SemanticErrorException: 'While parsing a block mapping, did not find expected key.' exception 

NOTE: I do have a kubeconfig file on my local hard drive in: C:\Users\my_name\.kube\config which I populated with exactly what the more ~/.kube/config command in PowerShell provided. This includes what looks like all the necessary certificate data.

Hoping to see if this remote access via this library is even possible ... thank you!

**Comment by tg123** [comment]
```
var config = new KubernetesClientConfiguration {  Host = "http://127.0.0.1:8080" };
```

**Reply by josha1971** [reply]
Thanks for the reply! So with the 8080 proxy setup via PowerShell after ssh'ing into my cluster, I added that line for the setting up the config object as you did there ... the code can get through that line now as well as newing up a Kubernetes object. However, when I try to run the 

client.CoreV1.ListNamespacePod("default") call, I get this: 
![image](https://user-images.githubusercontent.com/99064704/195613815-77854dab-8c1d-4e46-9ab2-d112ea2e2407.png)

**Reply by tg123** [reply]
a bit confused 
where do you want to run your client?

proxy? ssh?

could you please draw it out and i will help you understand what is the next step
